### PR TITLE
feat(tracing): add net.peer.name attribute if balancer_data.hostname is available

### PR DESCRIPTION
**Pull request by backjo**
This records the upstream hostname if available on the balancer spans.

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

Added new span attribute `net.peer.name` if balancer_data.hostname is available.

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Review by attenuation — COMMENTED** [review]
same as https://github.com/Kong/kong/pull/10723#issuecomment-1521073370, Please try to add some tests to this.

**Comment by StarlightIbuki** [comment]
> same as [#10723 (comment)](https://github.com/Kong/kong/pull/10723#issuecomment-1521073370), Please try to add some tests to this.

And also the changelog.

**Comment by oowl** [comment]
@backjo Any update here?

**Comment by backjo** [comment]
@hbagdi yep, cleaning it up rn

**Comment by backjo** [comment]
@hbagdi this is good to go I think

**Review by oowl — APPROVED** [review]

**Comment by oowl** [comment]
waiting for merged, And then I will do another cherry-pick-related job.

File: CHANGELOG.md [hunk 1]
```diff
@@ -88,6 +88,8 @@
   [#11082](https://github.com/Kong/kong/pull/11082)
 - Fix an issue where the router of flavor `expressions` can not configure https redirection.
   [#11166](https://github.com/Kong/kong/pull/11166)
+- Added new span attribute `net.peer.name` if balancer_data.hostname is available.
```
Comment (chronolaw) [inline CHANGELOG.md:91]:
```suggestion
- Added new span attribute `net.peer.name` if balancer_data.hostname is available.
  Thanks [@backjo](https://github.com/backjo) for contributing this change.
```

**Review by StarlightIbuki — APPROVED** [review]
LGTM.

**Pull request merged**